packages/askui-nodejs/src/execution/inference-client.ts

```diff
@@ -75,12 +81,18 @@ export class InferenceClient {
     const resizedImage = await this.resizeIfNeeded(customElements, image);
     const response = await this.httpClient.post<InferenceResponseBody>(
       this.urls.inference,
-      {
-        customElements,
-        image: resizedImage.base64Image,
+      this.urls.inference.endsWith('predict') ? {
+        // V4 only supports the base64 image and not the metadata of the base64 string
+        image: resizedImage.base64Image?.split(',')[1],
         instruction,
-        modelComposition: this.modelComposition,
-      },
+        tasks: ['OCR'],
```

Comment:
This will lead to only detecting text elements? Is this the intended behavior?

Reply:
Yes.

We moved away from Triton based model inference to a container based deploytment which serves only 1 TASK (initially started off as OCR but backend is build in a way which is easily extendable).

You can find the details in this [EPIC](https://askyourui.atlassian.net/browse/ML-463) and its related Stories.

You can find other implementations:
- Backend Implementation in this [PR](https://github.com/askui/askui-inference/pull/186).
- Infrastructure in this [folder](https://github.com/askui/infrastructure/tree/main/clusters/askui-dev-v2/default/end-to-end-ocr)
- Desktop Application for Flutter in this [Repository](https://github.com/askui/ml-ocr-text-recognition-desktop-frontend)


Also the working solution excluding the Askui-lib can be found [here](https://www.notion.so/askui/007_Dev_ML_Update-a8236e2a690f4e9f9bf202eb1f57871a?pvs=4#b2f06ddfa5204c65bdc4ac5e8cc57e43)

Reply:
You the Inference [PR](https://github.com/askui/askui-inference/pull/186) if you want.

The only thing remaining is to move Threading to ThreadPool executor which should be completed today.

Reply:
@mlikasam-askui Move threading to Threadpool executor as done in the main branch in "detectors_event_pipes.py". Deployed changes to Infrastructure (without merging).

Reply:
Please correct me if I'm mistaken, but from what I understand:

Currently, users are encountering issues with OCR accuracy. To address this, you've implemented a solution to correct OCR inaccuracies, with the intention of allowing users to utilize the newly trained OCR.
However, there's a current limitation where users are restricted to only using text elements because the inference graph doesn't include other models. Consequently, any previous tests involving elements other than text, such as clicking on buttons, will fail.
Additionally, there's a constraint where only one OCR version is allowed per workspace ID.
 
For the "is-image-required" endpoint, if you're just moving the mouse to coordinates like (x=2, y=2), there's no need for an image or OCR. That's why the endpoint exists—to skip unnecessary image taking and sending.

What do you think about keeping the current inference graph setup, but swapping out the OCR with a trained one if it exsits? This way, we can minimize changes in the ADK since it still operates under the same logic.

Reply:
Yes, You understood most things correctly.

1.) Thanks for the explanation on the "is-image-required" endpoint. Fixed it.

2.) Any previous test will continue to work as they did. The current solution is intended for backward compatibility. This is because:
- There are 2 separate API versions. V3 consists of everything that was done previously and no changes have been implemented.
- The default version used for every initialization of the test (including previous tests written) will point to "V3" implementation.
- Only the new implementation where we specify the "inferenceServerApiVersion" as "v4" will go through the new process.

3.) Yes, The constraint for 1 trained model per workspace is intended. Every workspace will have its separate trained model. We are not training the base model for all the users right now. In later stages, we can use the data provided by the users to train the main model and update the base model if necessary.

4.) We initially did release a version to [prod](https://ml-ocr.askui.com/) where we had a streamlit application doing the same and the new model was being served from triton which was workspace specific. There is a limitation of triton that when we update the model e.g. the draft model it still did not update in inference because triton caches the initially loaded model from S3. And if we update the draft model it does not get reflected until we restart triton. This is why @MichaelMayer-askui added restarting the triton server every night [here](https://github.com/askui/infrastructure/blob/main/clusters/askui-prod-v2/default/triton-inference-server-gpu-cron.yaml) which is not optimal.
- The process of why we do things this way is because we want to get rid of triton and move everything to a containerized-based approach in which 1 container is responsible for 1 Task (OD, OCR...). Can be found in [This Epic](https://askyourui.atlassian.net/browse/ML-439) followed by [This Epic](https://askyourui.atlassian.net/browse/ML-463) in detail.
- This is just the first step in moving all models to the containerized-based approach.
- In the next steps we also want to add a new Decision Logic other than UIDT1. On Runtime the User will be able to use Different Decision logic to what feels best for the user's needs e.g. [PTA-TEXT](https://askyourui.atlassian.net/browse/ML-467)


Let me know if you want to schedule a call and make things more clearer.